app/code/Transoft/Blog/ViewModel/Bloglist.php

```diff
@@ -0,0 +1,105 @@
+<?php
+declare(strict_types=1);
+namespace Transoft\Blog\ViewModel;
+
+use Magento\Catalog\Block\Product\View;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\UrlInterface;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Transoft\Blog\Model\ResourceModel\Post\Collection;
+use Transoft\Blog\Model\ResourceModel\Post\CollectionFactory as PostCollectionFactory;
+
+/**
+ * Class Bloglist
+ *
+ * Blog list view model class
+ */
+class Bloglist implements ArgumentInterface
+{
+    /**
+     * @var PostCollectionFactory
+     */
+    protected $postCollectionFactory;
+
+    /**
+     * ScopeConfigInterface $scopeInterface
+     */
+    private $scopeInterface;
+
+    /**
+     * UrlInterface $urlInterface
+     */
+    private $urlInterface;
+
+    /**
+     * Constructor.
+     *
+     * @param PostCollectionFactory $postCollectionFactory
+     * @param UrlInterface $urlInterface
+     * @param ScopeConfigInterface $scopeInterface
+     */
+    public function __construct(
+        PostCollectionFactory $postCollectionFactory,
+        UrlInterface $urlInterface,
+        ScopeConfigInterface $scopeInterface
+    ) {
+        $this->postCollectionFactory = $postCollectionFactory;
+        $this->urlInterface = $urlInterface;
```

Comment:
Please rename field and argument to $urlBuilder

Reply:
renamed